### PR TITLE
Add parentName and parentTriggerValues attributes for templates

### PR DESCRIPTION
**Pull request by Amar3tto**

### CODE REVIEW

File: v1/src/main/java/com/google/cloud/teleport/spanner/ExportPipeline.java [hunk 1]
```diff
@@ -182,6 +182,8 @@ public interface ExportPipelineOptions extends PipelineOptions {
     @TemplateParameter.Text(
         order = 10,
         optional = true,
+        parentName = "shouldExportRelatedTables",
+        parentTriggerValues = {"true"},
```
Comment (an2x) [inline v1/src/main/java/com/google/cloud/teleport/spanner/ExportPipeline.java:186]: I think this needs to be reverted.  I don't know this template well, but the `shouldExportRelatedTables` parameter description says "If `tableNames` is specified but doesn't include related tables, this option must be set to true for a successful export." Which means in some cases `tableNames` parameter can be set to something with `shouldExportRelatedTables`  set to false. So we should show `tableNames` unconditionally.

File: v2/pubsub-to-redis/src/main/java/com/google/cloud/teleport/v2/templates/PubSubToRedis.java [hunk 2]
```diff
@@ -219,6 +219,8 @@ public interface PubSubToRedisOptions
     @TemplateParameter.Long(
         order = 8,
         optional = true,
+        parentName = "redisSinkType",
+        parentTriggerValues = {"HASH_SINK", "LOGGING_SINK"},
         description = "Hash key expiration time in sec (ttl)",
         helpText =
             "Key expiration time in sec (ttl, default for HASH_SINK is -1 i.e. no expiration)")
```
Comment (an2x) [inline v2/pubsub-to-redis/src/main/java/com/google/cloud/teleport/v2/templates/PubSubToRedis.java:226]: Can you please also update the description and mention this parameter applies only to HASH_SINK and LOGGING_SINK sink types? It's true, based on the implementation, but this is not clear from the parameter description (it won't be visible in the UI once we add parentTriggerValues anyway, but we use the parameter description to auto-generate docs in some other places).

**Review by an2x — CHANGES_REQUESTED** [review]

**Review by aksharauke — COMMENTED** [review]
The Spanner impacted templates will be removed once the PR is merged: https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/1413
So the changes can be skipped.

**Comment by akashorabek** [comment]
R: @an2x

File: v2/pubsub-to-redis/src/main/java/com/google/cloud/teleport/v2/templates/PubSubToRedis.java (hunk 2 above)
Comment (akashorabek) [reply v2/pubsub-to-redis/src/main/java/com/google/cloud/teleport/v2/templates/PubSubToRedis.java:226]: Done

**Comment by akashorabek** [comment]
> The Spanner impacted templates will be removed once the PR is merged: #1413 So the changes can be skipped.

@aksharauke, removed the changes from the files that will be deleted after [this PR](https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/1413) will be merged. Pls approve the changes.

**Review by an2x — APPROVED** [review]

**Review by damondouglas — CHANGES_REQUESTED** [review]
I see a spotless check error: https://github.com/GoogleCloudPlatform/DataflowTemplates/actions/runs/8788623511/job/24255216239?pr=1456#step:4:852

**Comment by akashorabek** [comment]
@damondouglas made a commit. Can you check again pls?

**Comment by akashorabek** [comment]
@damondouglas these test failures are new and was caused after [this PR](https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/1453) I guess.

**Pull request merged**